src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanSessionsService.kt

```diff
@@ -29,10 +29,10 @@ class ActionPlanSessionsService(
   val communityAPIBookingService: CommunityAPIBookingService,
   val appointmentRepository: AppointmentRepository,
 ) {
-  fun createUnscheduledSessionsForActionPlan(submittedActionPlan: ActionPlan) {
-    val numberOfSessions = submittedActionPlan.numberOfSessions!!
+  fun createUnscheduledSessionsForActionPlan(approvedActionPlan: ActionPlan) {
+    val numberOfSessions = approvedActionPlan.numberOfSessions!!
     for (i in 1..numberOfSessions) {
-      createSession(submittedActionPlan, i)
+      createSession(approvedActionPlan, i)
```

Comment:
Do you think we should have protection against duplicated sessions? I think this can only be solved by applying a uniqueness constraint on action_plan_id and session_number

Reply:
maybe - although for now it's not a concern because this is the only place action plan sessions are created, and they are done sequentially in this loop

Reply:
What if the user double clicks on the approval button?

Reply:
good point ! i've added a unique key in at the db level to protect against this scenario.